src/Service.Tests/Configuration/RuntimeConfigLoaderTests.cs

```diff
@@ -30,4 +33,73 @@ public async Task CanLoadStandardConfig(string configPath)
+    {
+        string fileContents = await File.ReadAllTextAsync(configPath);
+
+        // Parse the base JSON string
+        JObject baseJsonObject = JObject.Parse(fileContents);
+
+        // Create a new JArray to hold the values to be appended
+        JArray valuesToAppend = new(dataSourceFiles);
+
+        // Add or append the values to the base JSON
+        baseJsonObject.Add("data-source-files", valuesToAppend);
+
+        // Convert the modified JSON object back to a JSON string
+        string resultJson = baseJsonObject.ToString();
+
+        IFileSystem fs = new MockFileSystem(new Dictionary<string, MockFileData>() { { "dab-config.json", new MockFileData(resultJson) } });
+
+        FileSystemRuntimeConfigLoader loader = new(fs);
+
+        Assert.IsTrue(loader.TryLoadConfig("dab-config.json", out RuntimeConfig runtimeConfig), "Should successfully load config");
+        Assert.IsTrue(runtimeConfig.ListAllDataSources().Count() == 4, "Should have 4 data sources");
+        Assert.IsTrue(runtimeConfig.CosmosDataSourceUsed, "Should have CosmosDb data source");
+        Assert.IsTrue(runtimeConfig.SqlDataSourceUsed, "Should have Sql data source");
+        Assert.AreEqual(DatabaseType.CosmosDB_NoSQL, runtimeConfig.DataSource.DatabaseType, "Default datasource should be of root file database type.");
+    }
+
+    /// <summary>
+    /// Test validates that load fails when datasource files have duplicate entities.
+    /// Example: Publisher entity present in the 3 sql.json files.
+    /// </summary>
+    [DataTestMethod]
+    [DataRow("dab-config.CosmosDb_NoSql.json", new string[] { "dab-config.MsSql.json", "dab-config.MySql.json", "dab-config.PostgreSql.json" })]
+    public async Task FailLoadMultiDataSourceConfigDuplicateEntities(string configPath, IEnumerable<string> dataSourceFiles)
+    {
+        string fileContents = await File.ReadAllTextAsync(configPath);
+
+        // Parse the base JSON string
+        JObject baseJsonObject = JObject.Parse(fileContents);
+
+        // Create a new JArray to hold the values to be appended
+        JArray valuesToAppend = new(dataSourceFiles);
+
+        // Add or append the values to the base JSON
+        baseJsonObject.Add("data-source-files", valuesToAppend);
+
+        // Convert the modified JSON object back to a JSON string
+        string resultJson = baseJsonObject.ToString();
+
+        IFileSystem fs = new MockFileSystem(new Dictionary<string, MockFileData>() { { "dab-config.json", new MockFileData(resultJson) } });
+
+        FileSystemRuntimeConfigLoader loader = new(fs);
+
+        try
+        {
+            loader.TryLoadConfig("dab-config.json", out RuntimeConfig _);
+            Assert.Fail("Should not load config with duplicate entity names.");
+        }
+        catch (DataApiBuilderException ex)
+        {
+            Assert.IsTrue(ex.StatusCode == System.Net.HttpStatusCode.BadRequest);
```

Comment:
DAB server boot failure due to bad config generically leads to a 500 internal server error

Reply:
in this case I thought this would make more sense as in this case there is a client error of providing runtime configs with duplicate entity names.

Reply:
I see your point. Contextually, even though DABExceptions all explicitly reference an HTTP status code, the exceptions are really only useful when the engine has successfully started and a client can react to responses from DAB. If there is a configuration error that prevents dab from starting, it isn't really client addressable, and requires the developer to update the runtime config. 

I don't think we allow more than one request to the /config endpoints (hosted scenario) even when the config bad. So the error would not be client fixable, and would require either a restart and sending a new valid config, or restarting and providing a valid config from the file system.

-> 5XX error code would fit better here.

Reply:
Makes sense. changed to internal server error.